src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -56,7 +56,7 @@ public void WriteMetric(double value)
             Enqueue(value);
         }
 
-        public override string ToString() => $"EventCounter '{Name}' Count {_count} Mean {(_sum / _count).ToString("n3")}";
+        public override string ToString() => $"EventCounter '{Name}' Count {_count} Mean {{(_count == 0 ? "" : (_sum / _count).ToString("n3"))}}";
```

Comment:
Is it possible that _count could change concurrently, e.g. be reset from a non-zero to a zero value during the ToString call?

I'm wondering if this should instead be something like:
```suggestion
        public override string ToString()
        {
            int count = Volatile.Read(ref _count);
            return count == 0 ?
                $"EventCounter '{Name}' Count 0" :
                $"EventCounter '{Name}' Count {count} Mean {(_sum / count).ToString("n3")}";
        }
```

Reply:
Absolutely, I was wary of trying to get too smart about this, but if there's anyone's code I trust to be concurrency-safe it's yours.

Reply:
@sywhang would know best whether it's needed based on whatever guarantees we make on the type.  I just noticed that _count is used elsewhere while holding a lock, including in a method to reset it to 0, and it's not used in a lock here, hence my suggestion.  Alternatively a lock could also be used.  I'll leave it up to @sywhang what he'd like to see here.